toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -79,11 +121,72 @@ func compressWithZstd(partitionRawFilepath string) (partitionFilepath string, er
 		return "", fmt.Errorf("failed to compress %s with zstd:\n%w", partitionRawFilepath, err)
 	}
 
-	// Remove raw file since output partition format is raw-zst.
-	err = os.Remove(partitionRawFilepath)
+	return partitionRawFilepath + ".zst", nil
+}
+
+// Prepend a skippable frame with the metadata to the specified partition file.
+func addSkippableFrame(tempPartitionFilepath string, skippableFrameMetadata [SkippableFramePayloadSize]byte, partitionfileName string, outDir string) (partitionFilepath string, err error) {
+	// Open tempPartitionFile for reading
+	tempPartitionFile, err := os.OpenFile(tempPartitionFilepath, os.O_RDWR, PartitionFilePermissions)
```

Comment:
You can just use `os.ModePerm` (i.e. `0o777`) instead of `0o644`. The [umask](https://en.wikipedia.org/wiki/Umask) will also apply to the permissions and will ensure it is something sensible.

Reply:
Would I need to change the PartitionFilePermissions to 0777 too?

Reply:
Updated to use this, thanks!